File: ytsharp.client/src/app/youtube-dl/youtube-dl.model.ts
export interface DownloadRequest {
  id: string;
  url: string;
  audioOnly: boolean;
  options: string;
  outputFolder: string;
};
export interface DownloadStatus {
  id: string;
  url: string;
  state: string;
  progress: number;
  downloadSpeed: string;
  eta: string;
  output: string[];
  isCompleted: boolean;
  isSuccessful: boolean;
  filePath: string;
  errorMessage: string;
}

export interface VideoInfo {
  title: string;
  description: string;
  uploadDate: string;
  uploader: string;
  duration: number;
  viewCount: number;
  // Add other video metadata properties as needed
}

File: YtSharp.Server/Models/youtubeDlModel.cs
namespace YtSharp.Server.Models
{
    // Model to receive download requests
    public class DownloadRequest
    {
        public required string? Url { get; set; }
        public bool AudioOnly { get; set; }
        public string? Options { get; set; }
        public required string DownloadId { get; set; }
        public required string OutputFolder { get; set; }
    }

    // Model to track download status
    public class DownloadStatus
    {
        public string? Id { get; set; }
        public string? Url { get; set; }
        public string? State { get; set; }
        public double Progress { get; set; }
        public string? DownloadSpeed { get; set; }
        public string? ETA { get; set; }
        public List<string> Output { get; set; } = new List<string>();
        public bool IsCompleted { get; set; }
        public bool IsSuccessful { get; set; }
        public string? FilePath { get; set; }
        public string? ErrorMessage { get; set; }
    }
}

